test/specs/utils/client.spec.js

```diff
@@ -0,0 +1,119 @@
+import sinon from 'sinon';
+import hat from 'hat';
+
+import * as storage from '../../../src/frame/js/utils/storage';
+import * as clientUtils from '../../../src/frame/js/utils/client';
+import { __Rewire__ as ClientRewire } from '../../../src/frame/js/utils/client';
+
+describe('Client Utils', () => {
+    const sandbox = sinon.sandbox.create();
+    let appId;
+    let getItemStub;
+    let setItemStub;
+    let removeItemStub;
+
+    beforeEach(() => {
+        appId = hat();
+
+        getItemStub = sandbox.stub();
+        setItemStub = sandbox.stub();
+        removeItemStub = sandbox.stub();
+        ClientRewire('storage', {
+            ...storage,
+            getItem: getItemStub,
+            setItem: setItemStub,
+            removeItem: removeItemStub
+        });
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    describe('getClientId', () => {
+        it('should return legacy client id', () => {
+            const legacyId = hat();
+            getItemStub.withArgs('sk_deviceid').returns(legacyId);
+
+            const id = clientUtils.getClientId(appId);
+
+            id.should.eql(legacyId);
+
+            getItemStub.should.have.been.calledOnce;
```

Comment:
There are two different styles of this that are a bit more pretty/concise:

```
getItemStub.should.have.been.calledOnce.calledWith('sk_deviceid');
```

or, if you don't care about non-matching call count:

```
getItemStub.withArgs('sk_deviceid').should.have.been.calledOnce;
```

But in the end to each their own :D